src/FSharp.Analyzers/VirtualCallAnalyzer.fs

```diff
@@ -0,0 +1,203 @@
+            "Microsoft.FSharp.Collections.Seq.mapFold"
+            "Microsoft.FSharp.Collections.Seq.mapFoldBack"
+            "Microsoft.FSharp.Collections.Seq.map3"
+            "Microsoft.FSharp.Collections.Seq.mapi"
+            "Microsoft.FSharp.Collections.Seq.mapi2"
+            "Microsoft.FSharp.Collections.Seq.maxBy"
+            "Microsoft.FSharp.Collections.Seq.minBy"
+            "Microsoft.FSharp.Collections.Seq.permute"
+            "Microsoft.FSharp.Collections.Seq.pick"
+            "Microsoft.FSharp.Collections.Seq.reduce"
+            "Microsoft.FSharp.Collections.Seq.replicate"
+            "Microsoft.FSharp.Collections.Seq.reduceBack"
+            "Microsoft.FSharp.Collections.Seq.scan"
+            "Microsoft.FSharp.Collections.Seq.scanBack"
+            "Microsoft.FSharp.Collections.Seq.skip"
+            "Microsoft.FSharp.Collections.Seq.skipWhile"
+            "Microsoft.FSharp.Collections.Seq.sortWith"
+            "Microsoft.FSharp.Collections.Seq.sortBy"
+            "Microsoft.FSharp.Collections.Seq.sortByDescending"
+            "Microsoft.FSharp.Collections.Seq.sumBy"
+            "Microsoft.FSharp.Collections.Seq.take"
+            "Microsoft.FSharp.Collections.Seq.takeWhile"
+            "Microsoft.FSharp.Collections.Seq.tryFind"
+            "Microsoft.FSharp.Collections.Seq.tryFindBack"
+            "Microsoft.FSharp.Collections.Seq.tryFindIndex"
+            "Microsoft.FSharp.Collections.Seq.tryFindIndexBack"
+            "Microsoft.FSharp.Collections.Seq.tryItem"
+            "Microsoft.FSharp.Collections.Seq.tryPick"
+            "Microsoft.FSharp.Collections.Seq.transpose"
+            "Microsoft.FSharp.Collections.Seq.truncate"
+            "Microsoft.FSharp.Collections.Seq.unfold"
+            "Microsoft.FSharp.Collections.Seq.windowed"
+            "Microsoft.FSharp.Collections.Seq.zip"
+            "Microsoft.FSharp.Collections.Seq.zip3"
+            "Microsoft.FSharp.Collections.Seq.removeAt"
+            "Microsoft.FSharp.Collections.Seq.removeManyAt"
+            "Microsoft.FSharp.Collections.Seq.updateAt"
+            "Microsoft.FSharp.Collections.Seq.insertAt"
+            "Microsoft.FSharp.Collections.Seq.insertManyAt"
+            "Microsoft.FSharp.Collections.Seq.allPairs"
+        ]
+
+[<CliAnalyzer("VirtualCall Analyzer",
+              "Checks if calls of Seq functions can be replaced with functions from the collection modules")>]
+let virtualCallAnalyzer : Analyzer<CliContext> =
+    fun (ctx : CliContext) ->
+        async {
+            let state = ResizeArray<string * string * range> ()
+
+
+            let walker =
+                { new TypedTreeCollectorBase() with
+                    override _.WalkCall
+                        _
+                        (mfv : FSharpMemberOrFunctionOrValue)
+                        _
+                        _
+                        (args : FSharpExpr list)
+                        (range : range)
+                        =
```

Comment:
Could you add some comments on what the general flow is here?
This might be useful in a few weeks.